disk/disk.go

```diff
@@ -1 +1,97 @@
 package disk
+
+import (
+	"io"
+	"log"
+	"os"
+)
+
+// WriteAt writes len(p) bytes from p to the file at path at offset off.
+// It returns the number of bytes written from p (0 <= n <= len(p)) and
+// any error encountered that caused the write to stop early.
+// WriteAt must return a non-nil error if it returns n < len(p).
+// TODO(xiang90): []byte or io.reader?
+func WriteAt(path string, p []byte, off int64) (int, error) {
+	// block size
+	bsize := blockSize(path)
+	// payload size
+	psize := bsize - crc32Len
+
+	f, err := os.OpenFile(path, os.O_WRONLY, 0600)
+	if err != nil {
+		return 0, err
+	}
+	flen := getFileLength(f)
+
+	// offset index that starts to write in the next time
+	i := min(off, flen)
+	// number of bytes that has written
+	n := 0
+	// index that ends writing
+	end := off + int64(len(p))
+	rbuf := make([]byte, psize)
+	for i < end {
+		pidx := i / psize
+		poff := i - pidx*psize
+		wbuf := make([]byte, 0)
+
+		// read out the head of data in block
```

Comment:
well... i would 

1. find the index of the first block to write

2. find the index of the last block to write

3. if both first,last is in flen -> a

4. if first is in flen, but last is not -> b

5. if first is not in flen -> c

and  i would prefer to handle first,last out of for loop.